aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -216,7 +216,38 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
-
+std::tuple<Tensor&, Tensor&> cummax_out(Tensor& out, Tensor& indices, const Tensor& self, int64_t dim) {
+   {
+    NoNamesGuard guard;
+    if(self.dim() == 0) {
+      out.resize_({}).fill_(self.item());
+      indices.resize_({}).fill_(0);
+    }
+    else {
+      // update out_ and indices_ for the first values along the dimension dim
+      out.narrow(dim, 0, 1) = self.narrow(dim, 0, 1);
```

Comment:
don't you need to resize the outputs first?  (Like you do in the `dim() == 0` case?).

Reply:
yeah updated!